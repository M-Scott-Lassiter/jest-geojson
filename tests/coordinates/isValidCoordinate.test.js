// This matcher works on all the same valid2DCoordinate and valid3DCoordinate cases.
// Accordingly, those specific tests check that both functions work.
// This test suite checks the array elements sizing because 2D and 3D have different acceptance criteria.
// Finally, it tests the unique snapshots.

describe('Invalid Use Cases', () => {
    describe('Expect to fail with incorrect number of array elements:', () => {
        test.each([[[]], [[20]], [[20, 30, 0, 4]], [[20, 30, 0, 20, 30, 0, 20, 30, 0]]])(
            'expect(%p)',
            (badInput) => {
                expect(badInput).not.isValidCoordinate()
            }
        )
    })
})

describe('Error Snapshot Testing. Throws error:', () => {
    test('expect([0, 0]).not.isValidCoordinate', () => {
        expect(() => expect([0, 0]).not.isValidCoordinate()).toThrowErrorMatchingSnapshot()
    })

    test('expect([0, 0, 0]).not.isValidCoordinate', () => {
        expect(() => expect([0, 0, 0]).not.isValidCoordinate()).toThrowErrorMatchingSnapshot()
    })

    test('expect([0, 0]).isValidCoordinate', () => {
        expect(() => expect([0, 95]).isValidCoordinate()).toThrowErrorMatchingSnapshot()
    })

    test('expect([0, 0, 0]).isValidCoordinate', () => {
        expect(() => expect([0, 95, 0]).isValidCoordinate()).toThrowErrorMatchingSnapshot()
    })
})
