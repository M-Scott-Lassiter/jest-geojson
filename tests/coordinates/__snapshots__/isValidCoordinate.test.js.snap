// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Snapshot Testing. Throws error: expect([0, 0, 0]).isValidCoordinate 1`] = `
"[2mexpect([22m[31m[longitude, latitude, (altitude)][39m[2m).isValidCoordinate()[22m

Coordinate latitude must be a number between -90 and 90.

Received:  [31m[0, 95, 0][39m"
`;

exports[`Error Snapshot Testing. Throws error: expect([0, 0, 0]).not.isValidCoordinate 1`] = `
"[2mexpect([22m[31m[longitude, latitude, (altitude)][39m[2m).not.isValidCoordinate()[22m

Expected input to not be a two or three element array with longitude between (-90 to 90), 
        latitude between (-180 to 180), and (if a 3D coordinate) numeric altitude.

Received:  [31m[0, 0, 0][39m"
`;

exports[`Error Snapshot Testing. Throws error: expect([0, 0]).isValidCoordinate 1`] = `
"[2mexpect([22m[31m[longitude, latitude, (altitude)][39m[2m).isValidCoordinate()[22m

Coordinate latitude must be a number between -90 and 90.

Received:  [31m[0, 95][39m"
`;

exports[`Error Snapshot Testing. Throws error: expect([0, 0]).not.isValidCoordinate 1`] = `
"[2mexpect([22m[31m[longitude, latitude, (altitude)][39m[2m).not.isValidCoordinate()[22m

Expected input to not be a two or three element array with longitude between (-90 to 90), 
        latitude between (-180 to 180), and (if a 3D coordinate) numeric altitude.

Received:  [31m[0, 0][39m"
`;
